src/run/dependency/dependency-diagram.ts

```diff
@@ -0,0 +1,107 @@
+import { writeFileSync } from "fs";
+import AbstractDependencyGraph, { Logger, ServiceFinder, ServiceGitDescriptionAndOwner } from "./AbstractDependencyGraph.js";
+import DependencyNode from "../../model/DependencyNode.js";
+import { graphviz } from "node-graphviz";
+import open from "open";
+import Service from "../../model/Service.js";
+import { Inventory } from "../../state/inventory.js";
+
+type TraveseType = "service" | "system"
+
+export default class DependencyDiagram extends AbstractDependencyGraph {
```

Comment:
Please add some tests, there's a lot of complex logic here and digression would be easy in future